### ISSUE TITLE
Update .gitignore for Java template

### ISSUE THREAD

**Issue by jamestharpe** [tmobile/jazz#162]
### Description

Expand the default .gitignore for Java projects. Consider these entries:
https://github.com/github/gitignore/blob/master/Java.gitignore

### Steps to Reproduce

1. Create a new Java 8 project in Jazz
2. git clone the created repository
3. Note that the .gitignore isn't ignoring everything it should

**Expected behavior:** *.class and other ignorable files are ignored

**Actual behavior:** *.class and other ignorable files are not ignored

**Reproduces how often:** Every time

**Comment by suryajak** [comment]
Makes sense. We should add the respective ones for nodejs and python templates too.
